crates/swc_css_minifier/src/compress/selector.rs

```diff
@@ -0,0 +1,267 @@
+use swc_common::DUMMY_SP;
+use swc_css_ast::*;
+use swc_css_visit::{VisitMut, VisitMutWith};
+
+pub fn compress_selector() -> impl VisitMut {
+    CompressSelector {}
+}
+
+struct CompressSelector {}
+
+impl CompressSelector {}
+
+impl VisitMut for CompressSelector {
+    fn visit_mut_nth(&mut self, nth: &mut Nth) {
+        nth.visit_mut_children_with(self);
+
+        match &nth.nth {
+            // `2n+1` + `odd`
+            NthValue::AnPlusB(AnPlusB {
+                a: Some(a),
+                b: Some(b),
+                span,
+                ..
+            }) if *a == 2 && *b == 1 => {
+                nth.nth = NthValue::Ident(Ident {
+                    span: *span,
+                    value: "odd".into(),
+                    raw: "odd".into(),
+                });
+            }
+            // `even` + `2n`
+            NthValue::Ident(Ident { value, span, .. }) if &*value.to_lowercase() == "even" => {
+                nth.nth = NthValue::AnPlusB(AnPlusB {
+                    span: *span,
+                    a: Some(2),
+                    a_raw: Some("2".into()),
+                    b: Some(0),
+                    b_raw: Some("".into()),
+                });
+            }
+            // `0n+5` => `5`, `0n-5` => `-5`
+            NthValue::AnPlusB(AnPlusB {
+                a: Some(a),
+                b,
+                b_raw,
+                span,
+                ..
+            }) if *a == 0 => {
+                nth.nth = NthValue::AnPlusB(AnPlusB {
+                    span: *span,
+                    a: None,
+                    a_raw: None,
+                    b: *b,
+                    b_raw: b_raw.clone(),
+                });
+            }
+            // `2n+0` => `2n`, `-2n+0` => `-2n`
+            NthValue::AnPlusB(AnPlusB {
+                a,
+                a_raw,
+                b: Some(b),
+                span,
+                ..
+            }) if *b == 0 => {
+                nth.nth = NthValue::AnPlusB(AnPlusB {
+                    span: *span,
+                    a: *a,
+                    a_raw: a_raw.clone(),
+                    b: None,
+                    b_raw: None,
+                });
+            }
+            _ => {}
```

Comment:
you seem to lack cases for `a == 1` and `a == -1` too..

Reply:
`1` and `-1` covered by codegen, it is part of CSS serialization, `1` and `-1` can be safely removed without changing AST, i.e. `1n+4` == `n+4`